src/main/java/bio/overture/ego/security/OAuth2UserInfoTokenServices.java

```diff
@@ -60,7 +62,7 @@ public IDToken extractPrincipal(Map<String, Object> map) {
     if (map.get("email") instanceof String) {
       email = (String) map.get("email");
     } else {
-      return null;
+      throw new NoPrimaryEmailException("no_primary_email");
```

Comment:
This message is not useful, we already know there was no primary email from the error type. Maybe there are no relevant details to add but could still be proper readable language instead of this snake_case token.

Reply:
Yeah on my first go at this i'm just passing the message as a param to client side, but I can set that where the exception gets caught, and make this message more readable instead.